src/msftoolbox/modis/modis_client.py

```diff
@@ -0,0 +1,72 @@
+import requests
+
+
+class ModisClient:
+    """
+    A class to interact with the MODIS REST API for a specific product.
+    A list of MODIS products can be found here: https://modis.ornl.gov/rst/api/v1/products
+    """
+
+    MODIS_API_URL = """https://modis.ornl.gov/rst/api/v1/"""
+
+    def __init__(self, product, longitude, latitude):
+        self.product = product
+        self.longitude = longitude
+        self.latitude = latitude
+
+    def get_modis_product_dates(self):
+        """
+        MODIS releases data in 8-day cycles. This will return a list of valid dates in AYYYYDDD format.
+
+        Returns:
+            dict: A dictionary with a list of the valid MODIS and calendar dates.
+        """
+        date_url = f"{self.MODIS_API_URL}/{self.product}/dates?latitude={self.latitude}&longitude={self.longitude}"
+        response = requests.get(date_url)
+
+        response.raise_for_status()
```

Comment:
It is a design choice of course but I am quite interested why us raise for status rather than a try , except to catch the error? would you recommend this for our other classes as well? and what would be the best approach moving forward to have consistency across our toolbox?

Reply:
I followed what was in the other modules, if you'd prefer, I can update to try catch - we'll just have to decide what to do if the API fails, do we log and return empty values? We'll just have to document that.